### PR TITLE
react1 week2 homework

### PR DESCRIPTION
**Pull request by hamoudi-hamoudi**

### CODE REVIEW

File: react/react1/week2/src/components/Notes/Note.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, { useState } from "react";
+
+function Note(props) {
+  const { description, deadline, onDelete } = props;
+  const [isDone, setIsDone] = useState(false);
+
+  const handleClick = () => {
+    setIsDone((prev) => !prev);
+  };
```
Comment (YurijPaharev) [inline react/react1/week2/src/components/Notes/Note.jsx:9]: Probably you would want to move this functionality on a level higher, to control the state of list from the parent component

File: react/react1/week2/src/pages/NotePage.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  useEffect(() => {
+    const fetchTasks = async () => {
+      const response = await fetch(
+        "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+      );
+      const data = await response.json();
+      setTasks(data);
+    };
+    fetchTasks();
+  }, []);
```
Comment (YurijPaharev) [inline react/react1/week2/src/pages/NotePage.jsx:21]: Great! You can even do it without fetchTasks function, just adding async on the useEffect callback

File: react/react1/week2/src/pages/NotePage.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  const handleNoteValue = (e) => {
+    const { name, value } = e.target;
+    setNoteValue((prev) => ({ ...prev, [name]: value }));
+  };
```
Comment (YurijPaharev) [inline react/react1/week2/src/pages/NotePage.jsx:26]: Wonderful

File: react/react1/week2/src/pages/NotePage.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+          description: description,
+          deadline: deadline,
+        },
+      ]);
+      setNoteValue({
+        description: "",
+        deadline: "",
+      });
+    }
+  };
```
Comment (YurijPaharev) [inline react/react1/week2/src/pages/NotePage.jsx:48]: Nice!

File: react/react1/week2/src/pages/NotePage.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            deadline={e.deadline}
+            onDelete={() => deleteTodo(e.id)}
+          />
+        );
+      })
+    ) : (
+      <ul>
+        <p>No task found</p>
+      </ul>
+    );
```
Comment (YurijPaharev) [inline react/react1/week2/src/pages/NotePage.jsx:70]: For the optimization purposes, you may want to wrap this solution in useMemo hook

**Review by YurijPaharev — APPROVED** [review]
Nicely done! There are just a few comments that I left

**Comment by hamoudi-hamoudi** [comment]
Thank you so much for the review, i will go through the changes 💯

**Pull request merged**